packages/dataviews/src/types.ts

```diff
@@ -37,6 +37,11 @@ export interface Option< Value extends any = any > {
 	description?: string;
 }
 
+export type FieldElementsSource< Value extends any = any > =
+	| Option< Value >[]
+	| Promise< Option< Value >[] >
+	| ( () => Option< Value >[] | Promise< Option< Value >[] > );
+
```

Comment:
This could be simplified to:

```js
export type FieldElementsSource =
	| Option[]
	| Promise< Option[] >
	| ( () => Promise< Option[] > );
```

I was thinking that elements could be either `Option[]` or an async function, as the async function enables the framework to trigger retrieval when necessary. However, having the `Promise<Option[]>` as well enables consumers to trigger retrieval instead. Nothing against it, I am just curious if you have any more thoughts about this.

Reply:
> This could be simplified to:

About this, I simplified the type, but I have a question. We declear the type of `Option` like this

```
export interface Option< Value extends any = any > {
	value: Value;
	label: string;
	description?: string;
}
```

Using `Option[]` only, we lose the ability to declare the type of value. Is that expected?